disnake/guild.py

```diff
@@ -2005,6 +2006,96 @@ async def create_scheduled_event(
         data = await self._state.http.create_guild_scheduled_event(self.id, reason=reason, **fields)
         return GuildScheduledEvent(state=self._state, data=data)
 
+    async def welcome_screen(self) -> WelcomeScreen:
+        """|coro|
+
+        Retrieves the guild's :class:`WelcomeScreen`.
+
+        Requires the :attr:`~Permissions.manage_guild` permission if the welcome screen is not enabled.
+
+        .. note::
+
+            To know if the welcome screen is enabled check for the
+            presence of ``WELCOME_SCREEN_ENABLED`` in :attr:`Guild.features`
+
+        .. versionadded:: 2.5
+
+        Raises
+        ------
+        NotFound
+            The welcome screen is not set up, or you do not have permissions to view it.
+        HTTPException
+            Editing the welcome screen failed.
+
+        Returns
+        -------
+        :class:`WelcomeScreen`
+            The guild's welcome screen.
+        """
+        data = await self._state.http.get_guild_welcome_screen(self.id)
+        return WelcomeScreen(state=self._state, data=data, guild=self)
+
+    async def edit_welcome_screen(
+        self,
+        *,
+        enabled: bool = MISSING,
+        welcome_channels: List[WelcomeScreenChannel] = MISSING,
+        description: str = MISSING,
+        reason: Optional[str] = None,
+    ) -> WelcomeScreen:
+        """This is a lower level method to :meth:`WelcomeScreen.edit` that allows you
+        to edit the welcome screen without fetching it and save an API request.
+
+        .. versionadded:: 2.4
```

Comment:
this should be 2.5

Reply:
https://github.com/DisnakeDev/disnake/pull/339/commits/7dff3cffc54ae17be2b7b47848deacdd602da658